File: leetcode/string/1758.py
from typing import List
from queue import Queue
class Solution:
    def minOperations(self, s: str) -> int:
        ra:int = 0
        rb:int = 0
        ba:bool = True
        for ch in s:
            ra += ba ^ int(ch)
            rb += (not ba) ^ int(ch)
            ba = not ba
        return min(ra, rb)
if __name__ == "__main__":
    solution = Solution()
    grid = [['1','1','0','0','0'],
            ['1','1','0','0','0'],
            ['0','0','1','0','0'],
            ['0','0','0','1','1']]
    result = solution.minOperations("0100")
    print(f"result = {result}")
 

File: leetcode/string/1758.cpp
#include <algorithm>
#include <functional> //包含std::greater 和 std::less
#include <iostream>
#include <limits.h>
#include <math.h>
#include <queue>
#include <stack>
#include <unordered_set>
#include <vector>

using namespace std;
struct TreeNode {
  int val;
  TreeNode *left;
  TreeNode *right;
  TreeNode() : val(0), left(nullptr), right(nullptr) {}
  TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
  TreeNode(int x, TreeNode *left, TreeNode *right)
      : val(x), left(left), right(right) {}
};
struct ListNode {
  int val;
  ListNode *next;
  ListNode() : val(0), next(nullptr) {}
  ListNode(int x) : val(x), next(nullptr) {}
  ListNode(int x, ListNode *next) : val(x), next(next) {}
};
class Solution {
public:
  int minOperations(string s) {
    int ra = 0, rb = 0;
    bool ba = true;
    for (char ch : s) {
      ra += ba ^ (ch - '0');
      rb += !ba ^ (ch - '0');
      ba = !ba;
    }
    return min(ra, rb);
  }
};
int main() {
  // 示例二叉树
  Solution solution;
  vector<int> nums = {7, 1, 5, 3, 6, 4};
  int result = solution.minOperations("0100");
  std::cout << "result: " << result << std::endl;
}